### PR TITLE
Improve performance while initializing KafkaMetrics

### PR DESCRIPTION
**Pull request by kptfh**
https://github.com/micrometer-metrics/micrometer/issues/2415

### CODE REVIEW

**Review by shakuzen — COMMENTED** [review]
I don't see how this changes the iterations required. Grouping by meter name would need to be done outside of the `metrics.forEach` loop to reduce iterations, unless I'm missing something.

**Comment by kptfh** [comment]
Hi. Sure. It should be outside the loop. Inaccurately ported this change. Will fix.
Thank you for review.

**Comment by kptfh** [comment]
fixed

**Comment by jonatan-ivanov** [comment]
@kptfh It seems the compiler does not like it, could you please take a look?

**Comment by siim-romanov** [comment]
Probably just needs to catch up with this change in master
https://github.com/micrometer-metrics/micrometer/pull/2418

**Comment by kptfh** [comment]
@jonatan-ivanov @siim-romanov fixed

**Comment by kptfh** [comment]
Hi guys! Do you have any comments regarding this pr?

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java [hunk 1]
```diff
@@ -172,12 +177,15 @@ void checkAndBindMetrics(MeterRegistry registry) {
                 // Kafka has metrics with lower number of tags (e.g. with/without topic or partition tag)
                 // Remove meters with lower number of tags
                 boolean hasLessTags = false;
-                for (Meter other : registry.find(meterName).meters()) {
+                List<Meter> metersForName = registryMetersByNames.getOrDefault(meterName, emptyList());
+                List<Meter> metersWithLessTags = new ArrayList<>(metersForName.size());
+                for (Meter other : metersForName) {
```
Comment (jonatan-ivanov) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java:182]: Is `metersWithLessTags` and `metersForName` really used here?
How about doing this:
```suggestion
                for (Meter other : registryMetersByNames.getOrDefault(meterName, emptyList())) {
```

And also removing `metersWithLessTags.add(other);` and `metersForName.removeAll(metersWithLessTags);`

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

**Comment by jonatan-ivanov** [comment]
@kptfh Thanks for the fix, could you please rebase this to `1.6.x`?

**Review by jonatan-ivanov — APPROVED** [review]

**Pull request merged**